consensus/spos/bls/subroundStartRound.go

```diff
@@ -51,26 +66,62 @@ func NewSubroundStartRound(
 
 func checkNewSubroundStartRoundParams(
 	baseSubround *spos.Subround,
+	alarmScheduler core.TimersScheduler,
+	chanStopNodeProcess chan endProcess.ArgEndProcess,
 ) error {
 	if baseSubround == nil {
 		return spos.ErrNilSubround
 	}
 	if baseSubround.ConsensusState == nil {
 		return spos.ErrNilConsensusState
 	}
+	if check.IfNil(alarmScheduler) {
+		return spos.ErrNilAlarmScheduler
+	}
+	if chanStopNodeProcess == nil {
+		return spos.ErrNilEndProcessChan
+	}
 
 	err := spos.ValidateConsensusCore(baseSubround.ConsensusCoreHandler)
 
 	return err
 }
 
+func (sr *subroundStartRound) resetConsensusStuckAlarm() {
```

Comment:
why do we need this? Does the chronology implementation is not enough?

Reply:
Refactored.